### PR TITLE
Use UserEntityInterface on new password form.

### PR DESCRIPTION
**Pull request by demiankatz**
This fixes some new minor issues with missing UserEntityInterface calls discovered while testing #2233.

This also adds a Mink test for recover password functionality. This could be fleshed out, but it covers the bare minimum "happy path" for now.

### CODE REVIEW

File: module/VuFind/src/VuFind/Controller/MyResearchController.php [hunk 1]
```diff
@@ -2007,17 +2007,17 @@ public function verifyEmailAction()
      * already been loaded from an existing hash, this resets the hash and updates
      * the form so that the user can try again.
      *
-     * @param mixed     $userFromHash User loaded from database, or false if none.
-     * @param ViewModel $view         View object
+     * @param ?UserEntityInterface $userFromHash User loaded from database, or false if none.
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/Controller/MyResearchController.php]: Should be "null if none"

File: themes/bootstrap3/templates/Auth/AbstractBase/login.phtml [hunk 2]
```diff
@@ -19,7 +19,7 @@
         <a class="btn btn-link createAccountLink" href="<?=$this->url('myresearch-account') ?>?auth_method=<?=$account->getAuthMethod()?>"><?=$this->transEsc('Create New Account')?></a>
       <?php endif; ?>
       <?php if ($account->supportsRecovery()): ?>
-        <a class="btn btn-link" href="<?=$this->url('myresearch-recover') ?>?auth_method=<?=$account->getAuthMethod()?>"><?=$this->transEsc('Forgot Password')?></a>
+        <a class="btn btn-link recoverAccountLink" href="<?=$this->url('myresearch-recover') ?>?auth_method=<?=$account->getAuthMethod()?>"><?=$this->transEsc('Forgot Password')?></a>
```
Comment (EreMaijala) [inline themes/bootstrap3/templates/Auth/AbstractBase/login.phtml]: I'd use the dash-separated style for classes (`recover-account-link`).

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: themes/bootstrap3/templates/Auth/AbstractBase/login.phtml (hunk 2 above)
Comment (demiankatz) [reply themes/bootstrap3/templates/Auth/AbstractBase/login.phtml]: I did this for consistency with createAccountLink on the line above. I can change it, but then the question is whether I should change that one too. I think these are only really used for testing purposes, though I guess it's conceivable that somebody might use them for custom styling.

Comment (demiankatz) [reply themes/bootstrap3/templates/Auth/AbstractBase/login.phtml]: In any case, I've changed it as you suggest for now, but I'm leaving existing things alone. :-)

**Review by EreMaijala — APPROVED** [review]

File: themes/bootstrap3/templates/Auth/AbstractBase/login.phtml (hunk 2 above)
Comment (EreMaijala) [reply themes/bootstrap3/templates/Auth/AbstractBase/login.phtml]: Thanks! I think it's better this way. :)

**Pull request merged**